client/templates/userprofile/settings.js

```diff
@@ -1,151 +1,59 @@
-        var email = template.find("#email").value;
+});
 
-        if (Meteor.userId())
-        {
-            if (realname && username && gender && language && email && isEmail(email))
+Template.profileSettings.events({
+    "submit #passwordEdit": function (event) {
+        event.preventDefault();
+        //TODO Unshow error message after a while
+        oldPassword = AutoForm.getFieldValue('old', 'passwordEdit');
+        newPassword = AutoForm.getFieldValue('new', 'passwordEdit');
+        Accounts.changePassword(oldPassword, newPassword, function(error) {
+            if(error)
             {
-                Session.set("basicsuccess", "Data successfully changed!");
-                console.log("success");
-
-                //TODO simplify
-                Meteor.call("User.update", Meteor.userId(),"realname", realname);
-                Meteor.call("User.update", Meteor.userId(),"username", username);
-                Meteor.call("User.update", Meteor.userId(),"gender", gender);
-                Meteor.call("User.update", Meteor.userId(),"email", email);
-                Meteor.call("User.update", Meteor.userId(),"language", language);
-                Meteor.call("User.update", Meteor.userId(),"biography", biography);
-
+                $("#passwordError #passwordErrorMessage").text(error);
+                $("#passwordError").show(500, function() { $("#passwordError").delay(3000).hide(500)});
             }
-            else
-            {
-                console.log("there has been a problem!");
+            else{
+                $("#passwordSuccess #passwordSuccessMessage").text("You have successfully changed your password!");
+                $("#passwordSuccess").show(500, function() { $("#passwordSuccess").delay(3000).hide(500)});
             }
 
-        }
-
-            //Session.set("errors", "");
-            Meteor.call("User.update", Meteor.userId(),"realname", realname);
-
-
+        });
     },
-    'submit .changePassword': function(event,template) {
+    "submit #UserProfileEdit": function (event) {
         event.preventDefault();
-        var oldpassword = template.find("#oldPassword").value;
-        var password = template.find("#newPassword").value;
-        var passwordagain = template.find("#newPasswordCheck").value;
-        if ((password === passwordagain)) {
-            if (Meteor.userId()) {
-            Accounts.changePassword(oldpassword,password, function(error) {
-                if(error) {
-                    console.log("Something went wrong! " + error);
-
-                }
-                else {
-                    Session.set("perrors", "");
-                    Session.set("passwordsuccess","Password changed!");
-                }
-            });
-            }
-        }
-        else {
-            Session.set("perrors", "Your passwords do not match.")
-        }
+        $("#profileSuccess #profileSuccessMessage").text("You have successfully updated your profile!");
```

Comment:
Add this to an AutoForm callback, and use the sAlert package for simplicity. A [working example can be seen in the Apinf project](https://github.com/apinf/api-umbrella-dashboard/blob/1e87fdf2cb7b4ec121e95943260c1329fbdd4d4f/client/views/profile/profile.js#L2-L5).